src/dataset-builder/CohortEditor.ts

```diff
@@ -396,8 +396,19 @@ export const CriteriaGroupView: React.FC<CriteriaGroupViewProps> = (props) => {
                 Link,
                 {
                   'aria-label': 'delete group',
-                  onClick: () =>
-                    updateCohort(_.set('criteriaGroups', _.without([criteriaGroup], cohort.criteriaGroups))),
+                  onClick: () => {
+                    updateCohort(_.set('criteriaGroups', _.without([criteriaGroup], cohort.criteriaGroups)));
+                    if (cohort.criteriaGroups.indexOf(criteriaGroup) < cohort.criteriaGroups.length - 1) {
+                      // update names of all groups after it
+                      for (
```

Comment:
thanks for the comment. I wonder if there is a way to have this work automatically if we use indexPairs, @s-rubenstein do you know?

Reply:
Yes, there should be. I think you can do it with Lodash (or Remeda?). Maybe:
```
updateCohort(_.set('criteriaGroups', _.flow(
  _.without([criteriaGroup]),
  Utils.toIndexPairs,
  _.map([criteriaGroup, i] => _.set('name', `Group ${i + 1}`, criteriaGroup))
)(cohort.criteriaGroups)))
```

Because it's not a problem to overwrite the names of earlier ones, since that will still get us the list in order.

(Disclaimer, I didn't try this so it may need some tweaking)